pkg/controller/s3/bucket/CORSConfig.go

```diff
@@ -14,14 +14,13 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package bucketresources
+package bucket
```

Comment:
nitpick: I might have missed conversation on original PR, but was there a specific reason for adding these subresource clients to `controller` rather than `client` dir?

Reply:
I guess they could go to either one of them. They are very similar to `ExternalClient` with their interface which makes them closer to be in `controller` package but at the same time they are called by the main controller which is similar to other things in `clients` folder. So, I'm on the fence.

On another note, fix for https://github.com/crossplane/provider-aws/issues/368 will likely change the interface and a few things so we can discuss this there as well since the shape of the interface would affect where they go.

Reply:
It was previously in the client dir, and I moved it under the controller since each struct is responsible for maintaining the lifecycle of a specific sub-resource similar to the controller. From my perspective, it doesn't make that much sense to have it under the client, since the structs are responsible for reconciling in the state of their resource.